internal/pkg/vm/internal/runtime/runtime.go

```diff
@@ -8,27 +8,163 @@ import (
+type Epoch types.BlockHeight
+
+// Randomness is a string of random bytes
+// Review: placeholder for matching newtypes with the spec.
+type Randomness []byte
+
+// InvocInput are the params to invoke a method in an Actor.
+// Review: why is this a struct instead of arguments?
+type InvocInput struct {
+    To      address.Address
+    Method  types.MethodID
+    Params  MethodParams
+    Value   types.AttoFIL
+}
+
+// ExitCode is the exit code of a method executing inside the VM.
+// Review: do we have reserved codes?
+// TODO: this has type UVarint in the spec
+type ExitCode types.Int64
+
+// InvocOutput is waht os returned by an actor executing a method.
+// Review: multiple returns vs a return struct. There are some minor implications with swapping vm implementations written in another language.
+type InvocOutput struct {
+	ExitCode     ExitCode
+    ReturnValue  []byte
+}
+
+// MethodParam is the parameter to an Actor method already encoded for execution in the VM.
+type MethodParam []byte
+// MethodParams is a list of `MethodParam` to be passed into an Actor method.
+type MethodParams []MethodParam
+
 // Runtime defines the ABI interface exposed to actors.
 type Runtime interface {
+	// Review: this is not in the spec anymore
 	Message() *types.UnsignedMessage
+
+	// Review: the spec splits this up and brings the get/put to be first class
 	Storage() Storage
+
+	// Review: changed from arguments to a struct
 	Send(to address.Address, method types.MethodID, value types.AttoFIL, params []interface{}) ([][]byte, uint8, error)
+
+	// Review: this is not in the spec anymore
 	AddressForNewActor() (address.Address, error)
+
+ 	// Review: renamed
 	BlockHeight() *types.BlockHeight
+
+	// Review: this is not in the spec anymore
 	MyBalance() types.AttoFIL
+
+	// Review: this changed a bit, not sure why it cant be determined off the result of `Caller()`
 	IsFromAccountActor() bool
+
+	// Review: this is not in the spec anymore
 	Charge(cost types.GasUnits) error
+
+	// Review: this got renamed and gained an offset arg
 	SampleChainRandomness(sampleHeight *types.BlockHeight) ([]byte, error)
 
+	// unchanged
 	CreateNewActor(addr address.Address, code cid.Cid, initalizationParams interface{}) error
 
+	// Review: this is not in the spec anymore
 	Verifier() verification.Verifier
 }
 
 // Storage defines the storage module exposed to actors.
 type Storage interface {
+	Head() cid.Cid
 	Put(interface{}) (cid.Cid, error)
+	// Dragons: this interface is wrong, the caller does not know how the object got serialized (Put/1 takes an interface{} not bytes)
+	// TODO: change to `Get(cid, interface{}) error`
 	Get(cid.Cid) ([]byte, error)
+	// Review: why is this needed on the actor API? Actor commit is implicit upon a succesfull termination.
+	// Review: an explicit Commit() on actor code leads to bugs
+	// Review: what situation requires state to change, succesfully execute the method, but not to be commited?
```

Comment:
These are good points / questions.  cc @acruikshank

Reply:
I too think this is internal, not exposed to actors.